WordPress/gutenberg
# Components: Add lint rules for theme color CSS var usage

Part of #44116
Follow-up to #58098

## What?

Adds stylelint and eslint rules to prevent direct usage of `wp-components-color-*` CSS variables in the components package.

## Why?

To ensure proper fallback values and to keep the code DRY.

## How?

Although we are restricting variable access through `var()`, at the moment I would still like to allow overriding definitions (`--wp-components-color-foo: #000`). There are legitimate use cases for it, and for now it's helpful to understand what those needs are.

## Testing Instructions

1. In the first commit where the eslint rule is added (6ccb546a496856f9935ea43c1cafcab2bf39df84), run `npm run lint:js`. The violations should be caught.
2. In the next commit where the violations are fixed (94f6be31a3319ef489c8c87af2aa91d296319242), the linter should pass.
3. In the final commit where the stylelint rule is added (1a0dc3b0993a0bb12c4631bd27a3f7ec88739f6b), run `npm run lint:css`. The linter should pass.
4. Test that the stylelint rule is working by adding a violation in a wp-components SCSS file somewhere.
